src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/RazorLanguageVersion.cs

```diff
@@ -53,6 +54,7 @@ private static FrozenDictionary<string, RazorLanguageVersion> BuildKnownVersions
             ["7.0"] = Version_7_0,
             ["8.0"] = Version_8_0,
             ["9.0"] = Version_9_0,
+            ["10.0"] = Version_10_0,
```

Comment:
This feature needs to be tied to .NET 11 in order to not break CI builds, we just have to decide what Razor lang version that is. .NET 10 shipped with RazorLangVersion 9, sadly, and it's not clear to me if its too late to change that so everything lines up nicely:
https://github.com/dotnet/sdk/blob/main/src/RazorSdk/Targets/Sdk.Razor.CurrentVersion.targets#L57